File: sol/3105/3105.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        n = len(nums)
        inc = 1
        dec = 1
        res = 1
        for i in range(1, n):
            if nums[i] > nums[i - 1]:
                inc += 1
                dec = 1
                res = max(res, inc)
            elif nums[i] < nums[i - 1]:
                dec += 1
                inc = 1
                res = max(res, dec)
            else:
                dec = 1
                inc = 1
        return res

File: sol/3105/3105.cpp
struct Solution {
    int longestMonotonicSubarray(vector<int>& nums) {
        int n = nums.size(), inc = 1, dec = 1, res = 1;
        for (int i = 1; i < n; ++i){
            if (nums[i] > nums[i - 1]){
                dec = 1;
                res = max(res, ++inc);
            } else if (nums[i] < nums[i - 1]){
                inc = 1;
                res = max(res, ++dec);
            } else {
                inc = 1;
                dec = 1;
            }
        }
        return res;
    }
};
